### ISSUE TITLE
rerun formatter should have an option of formatting without line numbers pointing to specific scenarios, allowing entire feature to be rerun if one scenario fails

### ISSUE THREAD

**Issue by henrykuijpers** [cucumber/cucumber-jvm#2489]
### 🤔 What's the problem you're trying to solve?

We're using the feature file to contain multiple scenarios that need to be executed. Sometimes, a scenario is dependant on an earlier scenario within that feature file. Therefore, we would like to have the option of the rerun formatter not appending any line numbers to the rerun.txt file, so that, in our case, the mutations that an earlier scenario performs, are reflected in later and eventually in the failed scenario.

### ✨ What's your proposed solution?

I would like to have an option i.e. "rerunEntireFeature=true" to turn off the generating of line numbers in the rerun.txt

Or, if this is totally against the concept, I would like to hear what we should better do in our case.

### ⛏Have you considered any alternatives or workarounds?

We can make a custom post-processing step that removes the line numbers, but it would be nice to have this contained in cucumber-jvm.

We come from a situation where we had dependant scenarios across the entire features-folder. This caused us to need steps to identify these dependencies + a custom "CucumberWithOrder" (based on "Cucumber") to reorder the scenarios before they were loaded. We have removed this and we have instead gathered all the scenarios that are depending on each other in the same feature file, which is executed in that order.

### Additional context

Very open to better suggestions. :)

**Comment by mpkorstanje** [comment]
> We come from a situation where we had dependant scenarios across the entire features-folder

The fundamental unit of execution is a scenario and Cucumber does not facilitate dependent scenarios. However implementing a plugin, especially the rerun formatter is relatively uncomplicated. Have a look at:

https://github.com/cucumber/cucumber-jvm/blob/main/core/src/main/java/io/cucumber/core/plugin/RerunFormatter.java

**Comment by mpkorstanje** [comment]
> And even, when executing scenarios in parallel, the JUnit implementation even guarantees the execution of scenarios in the order they are defined in the feature file as well.

JUnit 4 in combination with Maven does not support more then 2 levels of paralism. (runner class and features). If you were to use Gradle you'd only get 1 level of paralism (runner class). 

With TestNG, JUnit 5 and the CLI scenarios are executed without consideration for the feature file they're in.

Do have a look at what our friends at Specflow wrote about fixing dependent scenarios properly:

https://specflow.org/challenges/chain-of-dependent-scenarios/

Alternatively merge all dependent scenarios into a single scenario.

> However, we'd like to have this functionality available OOTB. :)

You use case is broadly considered a bad pattern, so I see no need to provide out of the box support for it.
